### PR TITLE
Inline documentation of the criteo dataset

### PR DESCRIPTION
**Pull request by soma2000-lang**
## Description
I have tried to do the inline documentation of the criteo.py file in the repo.I have documented only some parts of it ,as I want to check first whether I am going in the right direction or not the rest required changes can be done after feedback

Issue #165 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/EnvisEdge/labels)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by soma2000-lang** [comment]
@souravrhythm  Please review my pr

File: fedrec/datasets/criteo.py [hunk 1]
```diff
@@ -4,20 +4,47 @@
...
 # pytorch
 import torch
-from torch.utils.data import Dataset
+from torch.utils.data import Dataset #importing all the necessary libraries
 
 
 class CriteoDataset(Dataset):
-    def __init__(
-            self,
+    """Class docstrings go here"""
```
Comment (divyanshugit) [inline fedrec/datasets/criteo.py:11]: You don't have to mention this one. Can you please check [this](https://numpydoc.readthedocs.io/en/latest/format.html)

File: fedrec/datasets/criteo.py [hunk 2]
```diff
@@ -4,20 +4,47 @@
 import numpy as np
 # pytorch
 import torch
-from torch.utils.data import Dataset
+from torch.utils.data import Dataset #importing all the necessary libraries
```
Comment (divyanshugit) [inline fedrec/datasets/criteo.py:7]: I don't think, you need to mention this as in it's obvious.

File: fedrec/datasets/criteo.py [hunk 3]
```diff
@@ -4,20 +4,47 @@
...
 class CriteoDataset(Dataset):
-    def __init__(
-            self,
+    """Class docstrings go here"""
+
+
+
+
+
+""""This class is used to represent the criteo dataset
```
Comment (divyanshugit) [inline fedrec/datasets/criteo.py:17]: It's a simple dataloader for criteo. You can check out [this](https://pytorch.org/docs/stable/data.html) to get the better understanding about PyTorch dataloaders.

**Comment by soma2000-lang** [comment]
@divyanshugit  thanks for reviewing  my pr .I will make the suggested changes .Is there anything else that I should  add the documentation of??

File: fedrec/datasets/criteo.py (hunk 3 above)
Comment (soma2000-lang) [reply fedrec/datasets/criteo.py:17]: ok

**Comment by divyanshugit** [comment]
@soma2000-lang Can you check any of the previous file where we've written the inline documentation and try to follow that or you can check out [this](https://numpydoc.readthedocs.io/en/latest/format.html)

**Comment by soma2000-lang** [comment]
@divyanshugit I do that

**Comment by divyanshugit** [comment]
> @divyanshugit thanks for reviewing my pr .I will make the suggested changes .Is there anything else that I should add the documentation of??

You can explain a bit more about these function:
- collate_wrapper_criteo_offset
- offset_to_length_converter
- collate_wrapper_criteo_length
- make_criteo_data_and_loaders
according to Numpy Style guide

**Comment by soma2000-lang** [comment]
@divyanshugit I have made some changes ,please check

**Comment by divyanshugit** [comment]
> @divyanshugit I have made some changes ,please check

As I said earlier, you need to add inline comments as per the Numpy Style guideline.

**Comment by soma2000-lang** [comment]
@divyanshugit Can you please tell me what the  "collate_wrapper_criteo_offset" does in this criteo.py?

**Comment by divyanshugit** [comment]
> @divyanshugit Can you please tell me what the  "collate_wrapper_criteo_offset" does in this criteo.py?

It's just collecting and combining the bias_value(offset) which will be used during model training.

**Comment by ramesht007** [comment]
@soma2000-lang any update on this?

**Comment by soma2000-lang** [comment]
@ramesht007 I will submit a pr today itself

**Comment by soma2000-lang** [comment]
@divyanshugit @ramesht007 I have submitted a pr but this branch is giving conflicts.Please help

**Comment by divyanshugit** [comment]
Sorry, for the delay @soma2000-lang. Have you tried to rebase the branch or creating a new branch? I can't help you unless I know the exact issue.

**Comment by soma2000-lang** [comment]
@divyanshugit I have not tried to create a new branch

**Comment by ramesht007** [comment]
Hey @soma2000-lang, 
try reabasing your current repo to the new code and try fixing it.
You can checkout this [blog](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github) by GITHUB to fix this issu.
thanks
